proj_utils/Validator.py

```diff
@@ -0,0 +1,201 @@
+import datetime
+import json
+from jsonschema.validators import Draft3Validator
+
+
+class Validator:
+    schema = {
+      "type": "object",
+      "properties": {
+        "studentDetails": {
+          "type": "object",
+          "properties": {
+            "firstName": {
+              "type": "string"
+            },
+            "lastName": {
+              "type": "string"
+            },
+            "id": {
+              "type": "integer"
+            }
+          },
+          "required": [
+            "firstName",
+            "lastName",
+            "id"
+          ]
+        },
+        "subjectGrades": {
+          "type": "array",
+          "items": [
+            {
+              "type": "object",
+              "properties": {
+                "subject": {
+                  "type": "string"
+                },
+                "grades": {
+                  "type": "array",
+                  "items": [
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    }
+                  ]
+                }
+              },
+              "required": [
+                "subject",
+                "grades"
+              ]
+            },
+            {
+              "type": "object",
+              "properties": {
+                "subject": {
+                  "type": "string"
+                },
+                "grades": {
+                  "type": "array",
+                  "items": [
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    },
+                    {
+                      "type": "integer"
+                    }
+                  ]
+                }
+              },
+              "required": [
+                "subject",
+                "grades"
+              ]
+            }
+          ]
+        },
+        "birthDate": {
+          "type": "string"
+        },
+        "age": {
+          "type": "integer"
+        },
+        "gender": {
+          "type": "string"
+        },
+        "behaviorGrade": {
+          "type": "integer"
+        },
+        "notes": {
+          "type": "string"
+        }
+      },
+      "required": [
+        "studentDetails",
+        "subjectGrades",
+        "birthDate",
+        "age",
+        "gender",
+        "behaviorGrade",
+        "notes"
+      ]
+    }
+
+    def __init__(self, body):
+        self.json_file = json.loads(body)
+        self.message = ""
+        self.is_valid_schema = Draft3Validator(Validator.schema).is_valid(self.json_file)
+
+    def is_valid_firstname(self, firstName):
+        if not firstName.isalpha():
+            self.message += "Invalid First Name !\n"
```

Comment:
this can be designed better. right now, in every function you add a log to the final message, but if you would want to change how it works, you would have to change every function